tock/tock/remote_user_auth.py

```diff
@@ -25,38 +24,32 @@ def email_to_username(email):
     return email_list[0][:30]
 
 
-class TockUserBackend(RemoteUserBackend):
+def verify_userdata(user):
+    """Ensure that authenticated users have associated `UserData` records.
+    """
+    try:
+        user = UserData.objects.get(user=user)
+    except UserData.DoesNotExist:
+        UserData.objects.create(
+            user=user,
+            start_date=datetime.date.today(),
+        )
 
-    def clean_username(self, email_address):
-        """
-        Performs any cleaning on the "username" prior to using it to get or
-        create the user object.  Returns the cleaned username.
 
-        By default, returns the username unchanged.
-        """
-        return email_to_username(email_address)
+class TockUserBackend(UaaBackend):
 
+    def create_user_with_email(email):
+        username = email_to_username(email)
 
-class EmailHeaderMiddleware(RemoteUserMiddleware):
-    header = 'HTTP_X_AUTH_USER'
+        try:
+            print("Try getting a user that exists already.")
+            user = User.objects.get(username=username)
+        except User.DoesNotExist:
+            print("Create a new user if they don't exist.")
```

Comment:
We'll want to clean up these `print` statements, or turn them into `logging` statements if they would still be useful (and, come to think of it, required for ATO purposes).